joinmarket/blockchaininterface.py

```diff
@@ -575,12 +575,37 @@ def sync_addresses(self, wallet):
             return
         log.debug('requesting wallet history')
         wallet_name = self.get_wallet_name(wallet)
+	#TODO It is worth considering making this user configurable:
         addr_req_count = 20
         wallet_addr_list = []
         for mix_depth in range(wallet.max_mix_depth):
             for forchange in [0, 1]:
+		#If we have an index-cache available, we can use it
+		#to decide how much to import (note that this list
+		#*always* starts from index 0 on each branch).
+		#In cases where the Bitcoin Core instance is fresh,
+		#this will allow the entire import+rescan to occur
+		#in 2 steps only.
+		if wallet.index_cache != [[0,0]]*5:
```

Comment:
Good catch, thanks. One of those "this'll do for a test, fix it up later..." things :)